src/react/components/SaleorProvider.tsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { ApolloProvider } from "@apollo/client/react";
+import { SaleorSDK } from "../../core";
+
+const SaleorContext = React.createContext<{ api: any; client: any } | null>(
+  null
+);
+
+export const SaleorProvider: React.FC<{ client: any }> = ({
+  client,
+  children,
+}) => {
+  const [context, setContext] = React.useState<{
+    api: any;
+    client: any;
+  } | null>(null);
+
+  React.useEffect(() => {
+    const api = SaleorSDK(client);
+    setContext({ api, client });
+  }, []);
+
+  if (client && context) {
+    return (
+      <SaleorContext.Provider value={null}>
+        <ApolloProvider client={client}>{children}</ApolloProvider>
```

Comment:
ApolloProvider is not needed, we should explicitly pass the apollo client to SaleorContext and extract it inside hooks:
```
function useAuthState(opts) {
  const { apolloClient } = useSaleorContext();

  return useQuery(query, { ...opts, client: apolloClient })
}
```